CONTRIBUTING.md

```diff
@@ -1,63 +1,124 @@
-6. After the review, you should resolve issues brought up by the reviewers as needed (pushing a new commit to address reviewers' comments), iterating until the reviewers give their thumbs up, the "LGTM" (acronym for "Looks Good To Me").
-7. Once the code has passed review the Pull Request can be merged into the distribution.
+Before starting to work on a feature or a fix, it's good practice to ensure that:
+
+  1. There is a ticket for your work in the project's [issue tracker](https://github.com/lampepfl/dotty/issues);
+  2. The ticket has been discussed and prioritized by the team.
+
+You should always perform your work in its own Git branch. The branch should be given a descriptive name that explains its intent. Some teams also like adding the ticket number and/or the [GitHub](http://github.com) user ID to the branch name, these details is up to each of the individual teams. (See below for more details on branch naming.)
+
+## Opening a PR
+
+  When the feature or fix is completed you should open a [Pull Request](https://help.github.com/articles/using-pull-requests) on GitHub. Please have a look at and follow the [Pull Request Policy](https://github.com/scala/scala/wiki/Pull-Request-Policy) for guidelines on submitting a pull request to the dotty project. (the pull request policy is the same as for the Scala programming language)
+
+## Identifying the kind of PR
+
+1. Documentation
+    Whether you finally decided you couldn't stand that annoying typo anymore, you fixed the outdated code sample in some comment, or you wrote a nice, comprehensive, overview for an under-documented package, some docs for a class or the specifics about a method, your documentation improvement is very much appreciated, and we will do our best to fasttrack it.
+
+    You can make these changes directly in your browser in GitHub, or follow the same process as for code. Up to you!
+
+2. Code
+
+    For bigger changes, we do recommend announcing your intentions on [gitter](https://gitter.im/lampepfl/dotty) first, to avoid duplicated effort, or spending a lot of time reworking something we are not able to change at this time in the release cycle, for example.
+
+
+3. Bug Fix
+
+    At the end of the commit message, include "Fixes dotty/bug#NNNN", where https://github.com/lampepfl/dotty/issues/NNNN tracks the bug you're fixing. We also recommend naming your branch after the ticket number.
+
+4. Enhancement or New Feature
+
+    For longer-running development, likely required for this category of code contributions, we suggest you include `topic/` or `wip/` in your branch name, to indicate that this is work in progress, and that others should be prepared to rebase if they branch off your branch.
+
+    Any language change (including bug fixes) must be accompanied by the relevant updates to the spec, which lives in the same repository for this reason.
```

Comment:
I think we don't have a fork of the spec in this repo? Maybe we should request language documentation in `docs/docs/`?